Libraries/Network/NetInfo.js

```diff
@@ -233,14 +230,25 @@ const NetInfo = {
     },
   },
 
-  isConnectionExpensive(callback: (metered: ?boolean, error?: string) => void): void {
+  isConnectionExpensive(): Promise {
+    if (arguments.length > 1) {
+      console.warn('NetInfo.isConnectionMetered(callback) is deprecated. Use the returned Promise instead.');
+      const callback = arguments[1];
+      if (Platform.OS === 'android') {
+        RCTNetInfo.isConnectionMetered().then(
+          metered => callback(metered),
+          error => callback(null, error)
+        );
+      } else {
+        callback(null, 'Unsupported');
+      }
+      return;
```

Comment:
By the way, thanks for your persistence in doing this. :)
It is very much appreciated.

Ok, here is an udea.
We can 

```
// decallbackifyHelper.js

// @deprecated, will be removed on March 1 2016
function promiseWithErrorFirstCallback(promise: Promise, callbacks: any[]): Promise {
  return promise.then((res) => {
    if (callbacks.length > 0) {
      callbacks[1](res);
    }
    return res;
  }, (error) => {
    if (callbacks.length > 0) {
      callbacks[0](error);
    }
    throw error;
  });
};

// @deprecated, will be removed on March 1 2016
function promiseWithSuccessFirstCallback(promise: Promise, callbacks: any[]): Promise {
  return promise.then((res) => {
    if (callbacks.length > 0) {
      callbacks[0](res);
    }
    return res;
  }, (error) => {
    if (callbacks.length > 0) {
      callbacks[1](error);
    }
    throw error;
  });
};
```

On March 1 we will run a simple code mod that will replace code like this:
```
const promiseWithSuccessFirstCallback = require('decallbackifyHelper').promiseWithSuccessFirstCallback;
return promiseWithSuccessFirstCallback(RCTNetInfo.isConnectionMetered(), [...arguments]);
```
into this
```
return RCTNetInfo.isConnectionMetered();
```